FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Extensions/ConfigurationRegistrationExtensions.cs

```diff
@@ -16,29 +19,20 @@ public static class ConfigurationRegistrationExtensions
             this IServiceCollection services,
             IConfiguration configuration)
         {
-            services.Configure<FhirServerConfiguration>(options =>
-                configuration.GetSection(ConfigurationConstants.FhirServerConfigurationKey).Bind(options));
-            services.Configure<JobConfiguration>(options =>
-                configuration.GetSection(ConfigurationConstants.JobConfigurationKey).Bind(options));
-            services.Configure<FilterConfiguration>(options =>
-                configuration.GetSection(ConfigurationConstants.FilterConfigurationKey).Bind(options));
-            services.Configure<FilterLocation>(options =>
-                configuration.GetSection(ConfigurationConstants.FilterConfigurationKey).Bind(options));
-            services.Configure<DataLakeStoreConfiguration>(options =>
-                configuration.GetSection(ConfigurationConstants.DataLakeStoreConfigurationKey).Bind(options));
-            services.Configure<ArrowConfiguration>(options =>
-                configuration.GetSection(ConfigurationConstants.ArrowConfigurationKey).Bind(options));
-            services.Configure<JobSchedulerConfiguration>(options =>
-                configuration.GetSection(ConfigurationConstants.SchedulerConfigurationKey).Bind(options));
-            services.Configure<SchemaConfiguration>(options =>
-                configuration.GetSection(ConfigurationConstants.SchemaConfigurationKey).Bind(options));
-            services.Configure<HealthCheckConfiguration>(options =>
-                configuration.GetSection(ConfigurationConstants.HealthCheckConfigurationKey).Bind(options));
-            services.Configure<StorageConfiguration>(options =>
-                configuration.GetSection(ConfigurationConstants.StorageConfigurationKey).Bind(options));
+            string configVersionValue = configuration[ConfigurationConstants.ConfigVersionKey];
```

Comment:
Maybe check if this if null, or use **V1** by default?

Reply:
We don't allow default values here and configVersion must be present with the settings. Thus we can better diagnostic settings according to its version